store/src/main/java/com/dropbox/android/external/store4/MemoryPolicy.kt

```diff
@@ -31,24 +47,30 @@ class MemoryPolicy internal constructor(
 
     val hasMaxSize: Boolean = maxSize != DEFAULT_SIZE_POLICY
 
-    class MemoryPolicyBuilder {
+    val hasMaxWeight: Boolean = maxWeight != DEFAULT_SIZE_POLICY
+
+    class MemoryPolicyBuilder<Key : Any, Value : Any> {
         private var expireAfterWrite = DEFAULT_DURATION_POLICY
         private var expireAfterAccess = DEFAULT_DURATION_POLICY
-        private var maxSize: Long = -1
+        private var maxSize: Long = DEFAULT_SIZE_POLICY
+        private var maxWeight: Long = DEFAULT_SIZE_POLICY
+        private var weigher: Weigher<Key, Value> = OneWeigher
```

Comment:
Store never had a memory policy for weigher, it was only an api in the cache artifact.  Seems fine to add